### PR TITLE
Remove `src/poetry/__init__.py` to convert to namespace package

### PR DESCRIPTION
**Pull request by lafrenierejm**
# Pull Request Check List

Resolves: #3735

- [ ] Added **tests** for changed code. (Not applicable)
- [ ] Updated **documentation** for changed code. (Not applicable)

The corresponding change was made in https://github.com/python-poetry/poetry-core/commit/fa93845ed48f8ba392da3628fd5718a84ebae5d6.

### CODE REVIEW

**Comment by lafrenierejm** [comment]
Any ideas why pre-commit is failing?

**Comment by dimbleby** [comment]
pre-commit is failing because it explicitly bans implicit-namespack-packages: https://github.com/python-poetry/poetry/blob/f33c9db2bbc083f3147dec0fc171df820baa4390/.pre-commit-config.yaml#L39.

If this MR goes ahead you'll also need to fix things that are broken by implicit namespaces.  mypy for instance, I think you'd want to set all of namespace_packages, explicit_package_bases, mypy_path - all as in poetry-core.

However, since poetry-core has already done this there is already no collision of `__init__.py`.  So perhaps #3735 can simply be closed without further action?

**Comment by lafrenierejm** [comment]
> If this MR goes ahead you'll also need to fix things that are broken by implicit namespaces.  mypy for instance, I think you'd want to set all of namespace_packages, explicit_package_bases, mypy_path - all as in poetry-core.

Thanks for the tip. I have resolved the errors that CI was raising. This PR _should_ now be mergeable if the maintainers so choose.

> However, since poetry-core has already done this - https://github.com/python-poetry/poetry-core/pull/251 - there is already no collision of `__init__.py`.  So perhaps #3735 can simply be closed without further action?

AFAIK you are correct that there's no longer any technical reason this PR would be required. My personal preference would still be to take ahead and make this change so the codebase is a proper PEP420 namespace package rather than implicitly relying on poetry-core's already having done so. If we don't proceed with these changes and a third package is ever added to the `poetry` namespace, the same errors would arise again.

File: pyproject.toml [hunk 1]
```diff
@@ -124,6 +125,9 @@ enable_error_code = [
     "redundant-expr",
     "truthy-bool",
 ]
+namespace_packages = true
+mypy_path = "src"
```
Comment (neersighted) [inline pyproject.toml:129]: I'd move mypy_path to be next to files, and put namespace_packages and explicit_package_bases right below it. After that, LGTM.

**Review by neersighted — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (lafrenierejm) [reply pyproject.toml]: Force-pushed https://github.com/python-poetry/poetry/commit/4e32bade3c76fca133f7a82795f686dedc3bc679 with these changes.

**Review by neersighted — APPROVED** [review]

**Pull request merged**